docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+existing Invenio installations.
+
+Module release
+--------------
+
+How to release an Invenio module.
+
+**Python module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.rst`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``version.py`` (follow semantic versioning).
+5. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+6. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+7. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
+
+**JavaScript/React module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.md`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``package.json`` (follow semantic versioning).
+5. Regenerate the ``package-lock.json``:
+    - ``rm package-lock.json``
+    - ``npm install``
+6. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+7. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+8. Regularly check ``Travis`` and ``NpmJS`` to ensure the release went through.
+
+**Backport changes to a maint release**
+
+When you need to create a new release for a maintained previous release,
+you will have to do the extra step of backporting the changes to the
+maintenance branch.
+
+After the procedure described above, you will have to:
+
+1. Checkout the ``maint`` branch and create a new release branch:
+    - ``git checkout maint-<major>.<minor>``
+    - ``git checkout -b rel-v<major>.<minor>.<patch>``
+2. Cherry-pick the commits that you need to backport, resolve any conflict.
+   (yes, cherry-pick :))
+3. Run tests.
+4. Createa new commit with updated ``changes`` file and bumped version,
+   as described above.
+5. Issue a pull request against the **maintenance branch** (
+   ``maint-<major>.<minor>``).
+6. If Travis fails:
+    - Fix issue and ensure head commit is the release commit (i.e. rebase if
+      necessary).
+7. Merge, tag and push release:
+    - ``git merge --ff-only rel-v<major>.<minor>.<patch>``
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware maint-<major>.<minor> v<major>.<minor>.<patch>``
```

Comment:
Do you have any backport at reach? We could add a link to it as an example.

Reply:
hmm by backport you mean github link? The commands will not be illustrated there, just a list of commits, no? Otherwise, I did not fully understood this.